architecture.md

```diff
@@ -2,165 +2,165 @@
+Since message posting does not permit the passing of functions (among other things), functions are serialized in the Component iframe container and deserialized in the outer window before rendering. This “serialization” of functions is better described as a transformation however, as any callbacks invoked in the outer window DOM must be wrapped in a call to postMessage in order for the method to be executed in the Component iframe container. Within the Component iframe container, handling requests to invoke a method involves using a key from the message to look up the callback to be executed.
 
 This implementation provides a functional, if limited, method for execution of callbacks across isolation boundaries, and currently supports:
 
-- Widgets executing callbacks entirely contained within the Widget
+- Components executing callbacks entirely contained within the Component
   - `<button onClick={() => State.update({ k: state.k + 1 })}>`
-- Child Widgets executing callbacks passed by their parent
+- Child Components executing callbacks passed by their parent
   - Parent passes a prop `onUpdate={() => State.update({ k: state.k + 1 })}` to its child
   - Child renders a button `<button onClick={props.onUpdate}>`
-  - When the button is clicked on the child Widget in the outer window, the `() => State.update({ k: state.k + 1 })` function executes in the parent Widget iframe container
-- Parent Widgets executing callback arguments to a method passed to the child Widget
+  - When the button is clicked on the child Component in the outer window, the `() => State.update({ k: state.k + 1 })` function executes in the parent Component iframe container
+- Parent Components executing callback arguments to a method passed to the child Component
   - Parent passes a prop `executeChildMethod={(cb) => cb()}` to its child
   - Child invokes the prop callback `props.executeChildMethod(() => State.update({ k: state.k + 1 }))`
-  - Parent executes the child callback, but only if the child callback is executed within the parent’s function (see below). **This appears to be the same level of support offered by the current Viewer however - it can execute the function argument from the child Widget in the props function, but cannot defer it. See [andyh.near/widget/RenderTestChild](https://near.social/#/mob.near/widget/WidgetSource?src=andyh.near/widget/RenderTestChild).**
+  - Parent executes the child callback, but only if the child callback is executed within the parent’s function (see below). **This appears to be the same level of support offered by the current Viewer however - it can execute the function argument from the child Component in the props function, but cannot defer it. See [andyh.near/component/RenderTestChild](https://near.social/#/mob.near/component/ComponentSource?src=andyh.near/component/RenderTestChild).**
```

Comment:
```suggestion
  - Parent executes the child callback, but only if the child callback is executed within the parent’s function (see below). **This appears to be the same level of support offered by the current Viewer however - it can execute the function argument from the child Component in the props function, but cannot defer it.**
```

The change in the path/URL breaks the link, but kind of a moot point since I must have deleted that component since writing this 😂

Reply:
haha I caught the other cases where find and replace affected links but missed that one 😅